mlflow/data/digest_utils.py

```diff
@@ -18,12 +18,16 @@ def compute_pandas_digest(df) -> str:
     import numpy as np
     import pandas as pd
 
-    # drop object columns
-    df = df.select_dtypes(exclude=["object"])
-
     # trim to max rows
     trimmed_df = df.head(MAX_ROWS)
 
```

Comment:
Note: intentionally rearranged ordering here to trim first, since we now have to iterate through the rows to identify if a column is a string column